auth/src/main/java/com/firebase/ui/auth/data/model/Resource.java

```diff
@@ -65,6 +65,10 @@ public T getValue() {
         return mValue;
     }
 
+    public boolean hasValue() {
+        return mValue != null;
+    }
```

Comment:
Please remove this. It's not used anywhere and you should be checking the states instead.

Reply:
Oops, done.